pkg/image/registry/imagestreamimage/rest.go

```diff
@@ -64,24 +67,40 @@ func (r *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
 		return nil, errors.NewNotFound("imageStreamImage", imageID)
 	}
 
+	set := util.NewStringSet()
 	for _, history := range repo.Status.Tags {
 		for _, tagging := range history.Items {
-			if tagging.Image == imageID {
-				image, err := r.imageRegistry.GetImage(ctx, imageID)
-				if err != nil {
-					return nil, err
-				}
-				imageWithMetadata, err := api.ImageWithMetadata(*image)
-				if err != nil {
-					return nil, err
-				}
-				isi := api.ImageStreamImage{Image: *imageWithMetadata}
-				isi.Namespace = kapi.NamespaceValue(ctx)
-				isi.Name = id
-				return &isi, nil
+			// TODO: trim based on a better algorithm
+			base := strings.TrimPrefix(tagging.Image, "sha256:")
```

Comment:
You can try to use `digest.ParseDigest`. If it succeeds, there are methods to get just the algorithm and just the hex parts separately. If it fails, it's not a v2 digest.